library/src/main/java/com/novoda/downloadmanager/lib/NotificationDisplayer.java

```diff
@@ -117,15 +118,22 @@ private void buildActionIntents(String tag, int type, Collection<DownloadBatch>
             builder.setAutoCancel(true);
 
             String action = DownloadStatus.isError(batch.getStatus()) ? Constants.ACTION_LIST : Constants.ACTION_OPEN;
-
-            Intent clickIntent = new Intent(action, uri, context, DownloadReceiver.class);
-            clickIntent.putExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_IDS, new long[]{batchId});
-            clickIntent.putExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_STATUSES, new int[]{batchStatus});
-            clickIntent.putExtra(DownloadReceiver.EXTRA_BATCH_ID, batchId);
+            Intent clickIntent = createClickIntent(action, batchId, batchStatus, uri);
             builder.setContentIntent(PendingIntent.getBroadcast(context, 0, clickIntent, PendingIntent.FLAG_UPDATE_CURRENT));
         }
     }
 
+    private Intent createClickIntent(String action, long batchId, int batchStatus, Uri uri) {
+        Intent clickIntent = new Intent(action, uri, context, DownloadReceiver.class);
+        clickIntent.putExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_IDS, new long[]{batchId});
+        clickIntent.putExtra(DownloadReceiver.EXTRA_BATCH_ID, batchId);
+
+        int status = statusTranslator.translate(batchStatus);
+        clickIntent.putExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_STATUSES, new int[]{status});
```

Comment:
we don't know. This was done already :-) and xavi pulled out a method here.

I think it might be to do with wanting / used to have grouped downloads on one notification